common/src/main/scala/org/neo4j/spark/converter/DataConverter.scala

```diff
@@ -0,0 +1,181 @@
+package org.neo4j.spark.converter
+
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.catalyst.expressions.{GenericRowWithSchema, UnsafeArrayData, UnsafeMapData, UnsafeRow}
```

Comment:
why do we need these `UnsafeXxx` types? that seems quite internal-ish?

Reply:
and `InternalRow` is not even compatible with `Row`... 🤕

Reply:
We're using these as they came from Spark, we are blocked here.

Reply:
I gave it a quick try, and `InternalRow` is indeed required, but you can use `ArrayData` instead of `UnsafeArrayData` as well as `MapData` instead of `UnsafeMapData`.
That will at least reduce the coupling with internal types (not to 0, but still).

Reply:
Sure I also did for `GenericRowWithSchema -> GenericRow`